### ISSUE TITLE
Remove flicker when scorlling

### ISSUE THREAD

**Issue by binaryplease** [drbig/snb#6]
When pressing the j and k (up and down scroll) keys the sceen flickers, due to beeing redrawn.
This is quite annoying, maybe there is a way to fix it.

**Comment by drbig** [comment]
Immediate "fix" (well, a workaround) I can think about is maybe a different terminal emulator - it doesn't flicker at all here with `urxvt`. But that might be due to other factors, so can't guarantee anything.

As for proper "fix" I remember I discovered very late into writing it that `ncurses` has pads (or pages?) so you can basically draw the whole thing and then just "move" the pad/page. But it was too late (would require pretty much a rewrite from scratch of the UI code). Feel free to look into it though (but be warned that `ncurses` is a dark territory ;)).

**Comment by binaryplease** [comment]
Hello Thanks for the quick reply.

I recorded a small demo of the problem you can find it here:
https://www.youtube.com/watch?v=kmq2dYI2YXE
On the video it is a bit hard to see, in real it flickers more, but you get an idea.

I am using urxvt as terminal emulator. After your suggestion I tried with a different one.
With lxterminal there is NO flicker at all. 

Is there any setting in rxvt that could cause this?
My Xressources looks like this:

https://gist.github.com/binaryplease/d0967acde687aec4b4dc

I'd be happy to contribute but have never developed in C only Java and Ruby.

**Comment by drbig** [comment]
Actually I *do* get flicker in `urxvt` just like you do in the video. Doesn't happen in either `xterm` or `sakura`. I guess I could hand-wave it as "bug in urxvt", which it might actually be (or not).

I run `urxvt` without any resources and it still flickered, so settings there don't seem to have any influence. Btw. I'm also running Arch so chances are we have everything at the same versions. I wonder if it flickers on other distros (different `urxvt` version, settings etc.).

I *hate* this kind of ghost-bugs.

**Comment by binaryplease** [comment]
After finding this post:
http://lists.schmorp.de/pipermail/rxvt-unicode/2007q4/000487.html
I realised it is not only snb that flickers with urxvt. 
I get the same results with nload, also nload doesn't flicker in lxterminal.
I really like urxvt as it works well with my tiling window manager but this is really annoying...

**Comment by drbig** [comment]
Yeah, my feeling was `urxvt` had been more stringent in some way - turns out it's fast enough to show you cleared screen between drawing... guess a case were too much speed is actually bad for the user.

I assume there are some incantations for `ncurses` that would help here (just clear internal buffer, not the 'physical' terminal, redraw in the internal buffer, flush it to terminal...) but that's low priority (at least for me). I'll mark this as a bug though.
